applicationinsights-react-native/src/ReactNativeBasePlugin.ts

```diff
@@ -0,0 +1,316 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import dynamicProto from "@microsoft/dynamicproto-js";
+import {
+    AnalyticsPluginIdentifier, IAppInsights, IDevice, IExceptionTelemetry, eSeverityLevel
+} from "@microsoft/applicationinsights-common";
+import {
+    BaseTelemetryPlugin, IAppInsightsCore, IConfigDefaults, IPlugin, IProcessTelemetryContext, IProcessTelemetryUnloadContext, ITelemetryItem,
+    ITelemetryPlugin, ITelemetryUnloadState, _eInternalMessageId, _throwInternal, _warnToConsole, arrForEach, dumpObj, eLoggingSeverity,
+    getExceptionName, onConfigChange
+} from "@microsoft/applicationinsights-core-js";
+import { getGlobal, strShimUndefined } from "@microsoft/applicationinsights-shims";
+import { INativeDevice, IReactNativePluginConfig } from "./Interfaces";
+import { isPromiseLike, isString, ITimerHandler, objDeepFreeze, scheduleTimeout } from "@nevware21/ts-utils";
+import { IDeviceInfoModule } from "./Interfaces/IDeviceInfoModule";
+
+declare var global: Window;
+const defaultReactNativePluginConfig: IConfigDefaults<IReactNativePluginConfig> = objDeepFreeze({
+    disableDeviceCollection: false,
+    disableExceptionCollection: false,
+    uniqueIdPromiseTimeout: 5000
+});
+
+export class ReactNativeBasePlugin extends BaseTelemetryPlugin {
```

Comment:
I was originally thinking that we would have a base class and then 2 implementation classes (the existing and one that didn't set the device info) -- but this works.

The only issue I can think of is that "generally" you would not expect to create and use "Base" classes instances in an application....

We can probably "fix" this by just changing the name of this class to something that doesn't contain Base (or doesn't read like it's a support class that should not be directly used).

Some possible options that come to mind (as we can't change the original without causing bigger breaking changes)

- ReactNativeDefaultPlugin
- ReactNativeNoDevicePlugin
- ReactNativePluginNoDeviceInfo
- ReactNativeManualDevicePlugin

etc, I'm open to other suggestions

Reply:
I like ReactNativeManualDevicePlugin~